src/ovms_py_tensor.cpp

```diff
@@ -0,0 +1,49 @@
+#include "ovms_py_tensor.hpp"
+
+#include <string>
+#include <vector>
+#include <pybind11/pybind11.h>
+#include <numeric>
+
+namespace py = pybind11;
+using namespace ovms;
+
+OvmsPyTensor::OvmsPyTensor(void *ptr, std::vector<py::ssize_t> shape, std::string datatype, size_t size) :
+    ptr(ptr),
+    shape(shape),
+    ndim(shape.size()),
+    format(),
+    itemsize(),
+    datatype(datatype),
+    size(size)
+{
+    if (datatypeToBufferFormat.count(datatype)) {
```

Comment:
You are searching in a map twice, once to check if key exists and then extracting value out of that key. Use .find to do it in one go

Reply:
```
auto it = datatypeToBufferFormat.find(datatype);
format = it != datatypeToBufferFormat.end() ?  it->second : "UINT8";
```

Reply:
And have UINT8 as a constexpr with descriptive name like `DEFAULT_PRECISION`